drivers: usb_dc_native_posix: do callback for ZLPs

ekohandel opened the pull request:
A Zero Length Packet can be used by higher layer stack to discover when an endpoint is being processed by the host. An example of this was introduced as part of 0127d000a287 ("usb: device: cdc_acm: Use ZLP to detect initial host read") in the CDC ACM class.

Not invoking the callback for ZLPs results in the higher layer stack not being informed when the packet is consumed. This manifests as a CDC ACM USB-IP device that cannot transmit to the host while being able to receive from the host.

ekohandel committed:
drivers: usb_dc_native_posix: do callback for ZLPs

A Zero Length Packet can be used by higher layer stack to discover when
an endpoint is being processed by the host. An example of this was
introduced as part of 0127d000a287 ("usb: device: cdc_acm: Use ZLP to
detect initial host read") in the CDC ACM class.

Not invoking the callback for ZLPs results in the higher layer stack not
being informed when the packet is consumed. This manifests as a CDC ACM
USB-IP device that cannot transmit to the host while being able to
receive from the host.

Signed-off-by: Abe Kohandel <abe.kohandel@gmail.com>

jfischer-no reviewed (changes requested):
There should not be Github reference in the commit message but reference to commit in the tree, like
...
introduced in the commit 0127d000a287
("usb: device: cdc_acm: Use ZLP to detect initial host read")
...

ekohandel commented:
> There should not be Github reference in the commit message but reference to commit in the tree, like ... introduced in the commit [0127d00](https://github.com/zephyrproject-rtos/zephyr/commit/0127d000a2871a2868067001ad55cfce40a4986e) ("usb: device: cdc_acm: Use ZLP to detect initial host read") ...

Sorry about that, I saw other examples of references to PRs and Issues and thought it was ok. I have now updated the reference to the commit itself.

jfischer-no commented:
> > There should not be Github reference in the commit message but reference to commit in the tree, like ... introduced in the commit [0127d00](https://github.com/zephyrproject-rtos/zephyr/commit/0127d000a2871a2868067001ad55cfce40a4986e) ("usb: device: cdc_acm: Use ZLP to detect initial host read") ...
> 
> Sorry about that, I saw other examples of references to PRs and Issues and thought it was ok. I have now updated the reference to the commit itself.

No problem, it is common practice to add a reference to the Github issue, mainly to allow Github automation to close it, there is no other benefit. Here, you want a reference to a commit in the tree that introduced the change. Thank you.
